packages/netlify-cms-core/src/actions/editorialWorkflow.ts

```diff
@@ -241,45 +249,42 @@ function unpublishedEntryDeleteError(collection, slug, transactionID) {
  * Exported Thunk Action Creators
  */
 
-export function loadUnpublishedEntry(collection, slug) {
-  return async (dispatch, getState) => {
+export function loadUnpublishedEntry(collection: Collection, slug: string) {
+  return async (dispatch: ThunkDispatch<State, {}, AnyAction>, getState: () => State) => {
     const state = getState();
     const backend = currentBackend(state.config);
     const entriesLoaded = get(state.editorialWorkflow.toJS(), 'pages.ids', false);
     //run possible unpublishedEntries migration
     if (!entriesLoaded) {
-      const response = await backend.unpublishedEntries(state.collections).catch(() => false);
-      response && dispatch(unpublishedEntriesLoaded(response.entries, response.pagination));
+      try {
+        const { entries, pagination } = await backend.unpublishedEntries(state.collections);
+        dispatch(unpublishedEntriesLoaded(entries, pagination));
+        // eslint-disable-next-line no-empty
+      } catch (e) {}
     }
 
     dispatch(unpublishedEntryLoading(collection, slug));
 
     try {
-      const entry = await backend.unpublishedEntry(collection, slug);
-      const mediaFiles = entry.mediaFiles;
+      const entry = (await backend.unpublishedEntry(collection, slug)) as EntryValue;
       const assetProxies = await Promise.all(
-        mediaFiles.map(({ file }) => createAssetProxy(file.name, file)),
-      );
-      dispatch(addAssets(assetProxies));
-      dispatch(
-        setDraftEntryMediaFiles(
-          assetProxies.map((asset, index) => ({
-            ...asset,
-            ...mediaFiles[index],
-            draft: true,
-          })),
-        ),
-      );
-      dispatch(
-        addMediaFilesToLibrary(
-          mediaFiles.map(file => ({
-            ...file,
-            draft: true,
-          })),
+        entry.mediaFiles.map(({ url, file, path }) =>
+          createAssetProxy({
+            path,
+            url,
+            file,
+          }),
         ),
       );
+      dispatch(addAssets(assetProxies));
 
-      dispatch(unpublishedEntryLoaded(collection, entry));
+      let mediaFiles: MediaFile[] = entry.mediaFiles.map(file => ({ ...file, draft: true }));
```

Comment:
The entry draft redux state will now always has all the relevant media files for that draft.
If the entry doesn't have a specific media folder we need to add the media files from the "global" folder.